Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,82 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private RDTreeManager _treeManager;
+    public float ResearchTimer { get; private set; }
+
+    public delegate void ResearchDoneEvent();
+    public ResearchDoneEvent ResearchDone { get; set; }
+
+    public ResearchNodeSetting NodeSetting {  get; private set; }
+
+    public SkillState CurrentSkillState { get;  set; }
+
+    private void Start()
+    {
+        NodeSetting = GetComponent<ResearchNodeSetting>();
+
+        ResearchDone += ResearchBought;
+        ResearchDone += SetNextStatesInTree;
+        ResearchDone += NodeSetting.UnlockObjects;
+        ResearchDone += _treeManager.ResearchFinished;
```

Comment:
All functions are internal, so there is no need for an event. Events are primarily used to communicate with other components without creating a dependency. For example, you can create an event for when a node is researched. This way, the audio manager can subscribe and play a sound when this event is triggered. Internally, we tend not to use events.